include/cst_tokenstream.h

```diff
@@ -77,7 +77,11 @@ typedef struct cst_tokenstream_struct {
     const cst_string *p_prepunctuationsymbols;
     const cst_string *p_postpunctuationsymbols;
 
+
     cst_string charclass[256];
+    cst_string *charclass2[256];
+    cst_string **charclass3[256];
+    cst_string ***charclass4[256];
 
```

Comment:
@forslund You can look at the [Unicode Character Database](http://www.unicode.org/Public/8.0.0/ucd/). The character classes (punctuation, etc.) are listed in the `UnicodeData.txt` file in the 3rd column. The values for these is defined in http://www.unicode.org/reports/tr44/#General_Category_Values. The counts for these as of Unicode 8.0 are (in alphabetical order of General Category):

| Count | General Category | Description |
|---|---|---|
| 65 | Cc | Control Codes |
| 150 | Cf | Format Control Character |
| 853859 | Cn | Unassigned |
| 137468 | Co | Private Use |
| 2048 | Cs | Surrogate Code Point (for UTF-16 surrogate pairs) |
| 1984 | Ll | Lower Case Letter |
| 248 | Lm | Letter Modifier |
| 105697 | Lo | Other Letter (e.g. Syllables and Ideographs) |
| 31 | Lt | Title Case Letter |
| 1631 | Lu | Upper Case Letter |
| 383 | Mc | Spacing Combining Mark |
| 13 | Me | Enclosing Combining Mark |
| 1567 | Mn | Non-Spacing Combining Mark |
| 550 | Nd | Decimal Digit |
| 236 | Nl | Letter-Like Numeric Character (e.g. Roman Numeral) |
| 647 | No | Other Numeric Character (e.g. Superscript Digits) |
| 10 | Pc | Connecting Punctuation (e.g. Tie Bar) |
| 24 | Pd | Dashes and Hyphens |
| 73 | Pe | Closing Punctuation (e.g. closing parenthesis) |
| 10 | Pf | Final (End) Quotation Mark |
| 12 | Pi | Initial (Start) Quotation Mark |
| 513 | Po | Other Punctuation |
| 75 | Ps | Opening Punctuation (e.g. open parenthesis) |
| 53 | Sc | Currency Symbol |
| 121 | Sk | Modifier Symbol |
| 948 | Sm | Mathematical Symbol |
| 5677 | So | Other Symbol |
| 1 | Zl | Line Separator |
| 1 | Zp | Paragraph Separator |
| 17 | Zs | Space Separator (Not Including `\r`, `\t`, etc. Control Characters) |